src/storage/core.rs

```diff
@@ -1228,7 +1233,7 @@ where
         }
     }
 
-    async fn fsyncdata(&self) -> IOResult<()> {
+    pub(crate) async fn fsyncdata(&self) -> IOResult<()> {
         self.safe.read().await.fsyncdata().await
```

Comment:
Additional check, that `dirty_bytes() > max_dirty_bytes_before_sync` is needed here to prevent unneeded call to `fsync`